.github/workflows/zephyr-hal.yml

```diff
@@ -66,11 +65,12 @@ jobs:
           # directory that contains the 'msdk' and 'hal_adi' folders.
           bash ./msdk/.github/workflows/scripts/zephyr-hal.sh
 
-      - name: Push hal_adi repository
-        uses: EndBug/add-and-commit@v9.0.1
-        with:
-          add: '*'
-          message: 'Updating Zephyr MSDK Hal based on msdk PR# $PR_NUM'
-          push: true
-          # Specify path in workspace to the repo you want to push to
-          cwd: './hal_adi'
+      - name: Push changes to hal_adi repository
```

Comment:
@sihyung-maxim still seems to be failing (see https://github.com/analogdevicesinc/msdk/actions/runs/9065017493/job/24904593346)

You may want to try adding the SSH link manually and explicitly pushing to it.  Should force git to use SSH

```shell
git remote add upstream git@github.com:analogdevicesinc/hal_adi.git
git fetch upstream --depth 1
git add *
git commit -m "Update Zephyr MSDK Hal based on MSDK PR: https://github.com/analogdevicesinc/msdk/pull/${PR_NUM}"
git push upstream
```